src/router.ts

```diff
@@ -22,14 +33,16 @@ const EXPERIMENTS_EXTRA_HEADERS: ReadonlyMap<HeaderKeys, string> = new Map([
     "default-src * blob: data:; script-src blob: https://cdn.ampproject.org/lts/ https://cdn.ampproject.org/rtv/ https://cdn.ampproject.org/sw/ https://cdn.ampproject.org/v0.js https://cdn.ampproject.org/v0.mjs https://cdn.ampproject.org/v0/ https://cdn.ampproject.org/viewer/; object-src 'none'; style-src 'unsafe-inline' https://cdn.ampproject.org/rtv/ https://cdn.materialdesignicons.com https://cloud.typography.com https://fast.fonts.net https://fonts.googleapis.com https://maxcdn.bootstrapcdn.com https://p.typekit.net https://pro.fontawesome.com https://use.fontawesome.com https://use.typekit.net; report-uri https://csp.withgoogle.com/csp/amp",
   ],
 ]);
+const CACHE_AFTER_SERVING_HEADERS = (url: string, cacheKey: string) =>
+  new Map([[HeaderKeys.X_CACHE_AFTER_SERVING, `${url};${cacheKey}`]]);
 
 const router = new Router();
 
 router.onerror = (req, res, status, error) => {
   if (!(error instanceof FetchError)) {
-    console.error(error);
     error = new FetchError(status ?? 500, 'Internal Server Error');
   }
+  console.error(error);
```

Comment:
Nit:
```suggestion
  console.error(error?.stack ?? error);
```

Reply:
Cloudflare's log reader actually expands the whole object, so console'ing just `error` is better here. e.g., that's the log message from trying to retrieve a file that doesn't exist:
![image](https://user-images.githubusercontent.com/1839738/155376718-98c7223b-f475-41a8-89b9-5dba0d42e418.png)

Reply:
Welp, turns out I was mistaken and your suggestion is better. Done :D